test/Hosting.CommandLine.Tests/HostBuilderExtensionsAttributeAPITests.cs

```diff
@@ -74,13 +74,36 @@ public async Task TestConfigureCommandLineApplication()
             Assert.NotNull(commandLineApp);
         }
 
+        [Fact]
+        public async Task TestUseCommandLineApplication()
+        {
+            CommandLineApplication<Return42Command> commandLineApp = default;
+            var hostBuilder = new HostBuilder();
+            hostBuilder.UseCommandLineApplication<Return42Command>(new string[0], app => commandLineApp = app);
+            var host = hostBuilder.Build();
+            await host.RunCommandLineApplicationAsync();
+            Assert.NotNull(commandLineApp);
+        }
+
+        [Fact]
+        public async Task UseCommandLineApplicationReThrowsExceptions()
+        {
+            var ex = await Assert.ThrowsAsync<InvalidOperationException>(
+                () => new HostBuilder()
+                    .ConfigureServices(collection => collection.AddSingleton<IConsole>(new TestConsole(_output)))
+                    .UseCommandLineApplication<ThrowsExceptionCommand>(new string[0], app => { })
+                    .Build()
+                    .RunCommandLineApplicationAsync());
+            Assert.Equal("A test", ex.Message);
+        }
+
         [Fact]
         public async Task ItThrowsOnUnknownSubCommand()
         {
             var ex = await Assert.ThrowsAsync<UnrecognizedCommandParsingException>(
                 () => new HostBuilder()
                     .ConfigureServices(collection => collection.AddSingleton<IConsole>(new TestConsole(_output)))
-                    .RunCommandLineApplicationAsync<ParentCommand>(new string[] { "return41" }));
```

Comment:
Why did so many of existing tests change to add `app => { }` to the `RunCommandLineApplicationAsync` call?

Reply:
I'm going to add a change to this PR to preserve the existing behavior. While not technically a binary breaking change, requiring this extra parameter would probably some users upgrading.

Reply:
Hey @natemcmaster - Thanks for taking the time to review this PR.

This was because there was code flagged as unreachable that I removed in several of the extension methods. Here's an [example](https://github.com/natemcmaster/CommandLineUtils/pull/415/files#diff-64279ca55776b3d59f8f6ee68e64094edbaa9ca7db7fbb9f8fd8f31d37b86a75L60).

After reviewing, I am now realizing that this is seemingly a breaking change and I shouldn't have made it.

Reply:
It looks like you've realized this too and already made the necessary changes to fix this behavior. Thanks.